src/nrnoc/seclist.cpp

```diff
@@ -110,23 +105,16 @@ static double children(void* v) {
     return 1.;
 }
 
-static Item* subtree1(List* sl, Section* sec) {
-    Item *i, *j, *last, *first;
-    Section* s;
-    /* it is confusing to span the tree from the root without using
-      recursion.
-    */
-    s = sec;
-    i = lappendsec(sl, s);
-    section_ref(s);
-    last = i->prev;
-    while (i != last) {
-        for (first = last->next, last = i, j = first; j->prev != last; j = j->next) {
-            s = hocSEC(j);
-            i = children1(sl, s);
-        }
+// Create a list of section from the subtree with root `sec` by
+// doing breadth-first traversal.
+static void subtree1(List* sl, Section* sec) {
+    const Item* end = sl;
+    Item* current = lappendsec(sl, sec);
```

Comment:
```suggestion
    // The pointer `sl` points to a past-the-end
    // marker. Therefore, appending items means
    // they appear immediately before `sl`.
    Item* current = lappendsec(sl, sec);
```

Reply:
I think the rewrite, and the two comments should be enough to enable future devs to figure it with reasonable ease (hopefully).